internal/controller/object/object.go

```diff
@@ -330,7 +330,7 @@ func (c *external) Update(ctx context.Context, mg resource.Managed) (managed.Ext
 		v1.LastAppliedConfigAnnotation: string(cr.Spec.ForProvider.Manifest.Raw),
 	})
 
-	if err := c.client.Apply(ctx, obj); err != nil {
+	if err := c.client.Update(ctx, obj); err != nil {
```

Comment:
Thanks for the PR, however, I am not sure if this is a good idea. 
AFAIR, some objects (like a k8s service) would always be updated and never synced since other controllers adding other fields and here we are reverting everything. So, the problem is bigger than it looks and I hope server side apply could help here (PR coming).

Reply:
Ok I'll wait for server-side apply. Any sort of ETA on that being delivered?

Reply:
this looks promising for the somewhat near future @bradkwadsworth-mw 😂 

https://github.com/crossplane-contrib/provider-kubernetes/pull/246